api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java

```diff
@@ -176,11 +176,35 @@ public CohortDefinition txTbNumeratorA() {
                 true));
     addGeneralParameters(i);
     cd.addSearch("i", map(i, generalParameterMapping));
+
     CohortDefinition ii = getInTBProgram();
     cd.addSearch("ii", map(ii, generalParameterMapping));
+
+    CohortDefinition patientswithPulmonaryTbDate =
+        TXTBQueries.getPatientsWithObsBetweenDates(
+            "Patients with Pulmonary TB Date",
+            hivMetadata.getPulmonaryTBConceptUuid(),
+            hivMetadata.getPatientFoundYesConcept(),
+            Arrays.asList(hivMetadata.getMasterCardEncounterType()));
+    cd.addSearch(
+        "patientswithPulmonaryTbDate", map(patientswithPulmonaryTbDate, codedObsParameterMapping));
+
+    CohortDefinition patientsWhoInitiatedTbTreatment =
```

Comment:
This definition looks like it has to exclude something look below
**all patients who started TB Treatment at any point in the previous 6 months before 
to the start date of the reporting period (Numerator for previous 6 months)**

Reply:
@ningosi  The dataset doesn't call that method directly, it rather calls `txTbNumerator()` which handles that exclusion.